src/neo/Network/P2P/TaskManager.cs

```diff
@@ -275,7 +275,7 @@ private void RequestTasks(TaskSession session)
                 for (uint i = currentHeight + 1; i <= headerHeight; i++)
                 {
                     UInt256 nextHash = Blockchain.Singleton.HeaderCache[i].Hash;
```

Comment:
```suggestion
                    UInt256 nextHash = NativeContract.Ledger.GetHeader(snapshot, i)?.Hash;
```
?

Reply:
If the header already exists in header cache we will not request for it. It might not be in snapshot. Maybe we should firstly check in cache then in snapshot.